PredixMobileUIKit/PredixBarChartView.swift

```diff
@@ -0,0 +1,330 @@
+    func createDataSets(bars: [Bar]) -> [BarChartDataSet] {
+        var dataSets: [BarChartDataSet] = []
+        for bar in bars {
+            var dataEntries: [BarChartDataEntry] = []
+            for i in 0 ..< bar.values.count {
+                let dataEntry = BarChartDataEntry(x: Double(i), y: bar.values[i])
+                dataEntries.append(dataEntry)
+            }
+            let chartDataSet = BarChartDataSet(values: dataEntries, label: bar.label)
+            if !bar.colors.isEmpty {
+                chartDataSet.colors = bar.colors
+            }
+            dataSets.append(chartDataSet)
+        }
+        return dataSets
+    }
+
+    /// Stacks all Bar DataSets in the Chart
+    internal func stackBarsOnChart() {
+        if !bars.isEmpty {
+
+            let groupCount = xAxisValues.count
+            let xaxis = xAxis
+            xaxis.valueFormatter = IndexAxisValueFormatter(values: xAxisValues)
+
+            let stackBarsDataSets = createDataSets(bars: bars)
+            let stackBarsCharData = BarChartData(dataSets: stackBarsDataSets)
+
+            let stackBarsStatingNumber = -0.5
+            xaxis.centerAxisLabelsEnabled = false
+            xaxis.axisMinimum = stackBarsStatingNumber
+            xaxis.axisMaximum = stackBarsStatingNumber + Double(groupCount)
+            data = stackBarsCharData
+        }
+    }
+
+    /// Groups all Bar DataSets in the Chart
+    internal func groupBarsOnChart() {
+        if !bars.isEmpty {
+
+            let groupCount = xAxisValues.count
+            let xaxis = xAxis
+            xaxis.valueFormatter = IndexAxisValueFormatter(values: xAxisValues)
+
+            let dataSets = createDataSets(bars: bars)
+            let groupBarsChartData = BarChartData(dataSets: dataSets)
+            let barCount = groupBarsChartData.dataSets.count
+            let barSpace = 0.05
+            let groupBarsStatingNumber = 0.0
+
+            /*
+             The equation to find the interval per "group" is:
+             (groupSpace * barSpace) * n + groupSpace = 1
+             Therefore by finding groupSpace we get:
+             groupSpace = 1 - numberOfBars * BarSpace / numberOfBars +1
+             */
+            let groupSpace = (1.0 - Double(barCount) * barSpace) / (Double(barCount) + 1.0)
+            let barWidth = groupSpace
+
+            groupBarsChartData.barWidth = barWidth
+            xaxis.centerAxisLabelsEnabled = true
+
+            xaxis.axisMinimum = groupBarsStatingNumber
+            let individualGroupSpace = groupBarsChartData.groupWidth(groupSpace: groupSpace, barSpace: barSpace)
+            xaxis.axisMaximum = groupBarsStatingNumber + individualGroupSpace * Double(groupCount)
+            groupBarsChartData.groupBars(fromX: Double(groupBarsStatingNumber), groupSpace: groupSpace, barSpace: barSpace)
+            data = groupBarsChartData
+        }
+    }
+
+    // MARK: PredixBarChartView  Handling Option Logic
+
+    /// Handle the options provided.
```

Comment:
Need more information in the code comment here. What kind of options are these? How is this used?